src/conf_mode/firewall.py

```diff
@@ -232,29 +231,28 @@ def verify_rule(firewall, rule_conf, ipv6):
         if side in rule_conf:
             side_conf = rule_conf[side]
 
-            if dict_search_args(side_conf, 'geoip', 'country_code'):
-                if 'address' in side_conf:
-                    raise ConfigError('Address and GeoIP cannot both be defined')
-
-                if dict_search_args(side_conf, 'group', 'address_group'):
-                    raise ConfigError('Address-group and GeoIP cannot both be defined')
-
-                if dict_search_args(side_conf, 'group', 'network_group'):
-                    raise ConfigError('Network-group and GeoIP cannot both be defined')
+            if {'address', 'fqdn', 'geoip'} <= set(side_conf):
+                raise ConfigError('Only one of address, fqdn or geoip can be specified')
```

Comment:
Does it check only if 3 options are defined?
What will be if there are 2 options? Raise config error says that only 1 option expected
Also, I think it should be uppercase `FQDN` and `GeoIP`. It is better to ask @dmbaturin

Reply:
Good point, this doesn't behave as I expected - I will fix it shortly.

As for case, @c-po previously mentioned we should avoid capitals in config definitions.